### PR TITLE
Update Docusaurus 3

### PR DESCRIPTION
**Pull request by Vrixyz**
Builds on https://github.com/dimforge/rapier.rs/pull/110

| docusaurus 2.4 | docusaurus 3 |
|--------|--------|
| docusaurus 2.4 | Screenshot 2024-07-05 at 09-58-37 Getting started Rapier |

### CODE REVIEW

**Comment by Vrixyz** [comment]
(not really related to this PR, but) on my linux, the `inject file` script seems to have a bug which omits the last `\n`, which makes yarn build fail ; I'll look into it, this might be the source of failure for CI

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,12 +10,17 @@ env:
...
     steps:
       - uses: actions/checkout@v4
-      - name: Run injection script locally
-        run: find docs/user_guides/templates/ -type f -print0 | OUTPUT_FOLDER=docs/user_guides/templates_injected xargs -0 ./inject_code_in_user_guides.sh
+      - name: User guides generation
+        run: ./generate_user_guides.sh
+      - name: yarn install
+        run: corepack enable && yarn
+      - name: yarn build
+        run: PUBLISH_MODE=1 yarn build
```
Comment (Vrixyz) [inline .github/workflows/ci.yml:22]: `yarn build` tested on CI is great, it currently outputs some warning concerning links, which might be related to https://github.com/dimforge/rapier.rs/issues/41

A good follow up Pull Request would be to eliminate these warnings. From my testing these warnings don´t indicate regressions.

File: docs-examples/inject_file/src/main.rs
Comment (Vrixyz) [inline docs-examples/inject_file/src/main.rs:1]: changes have been made to this script to make it more cross platform friendly. I tested on mac, windows, linux.

File: generate_user_guides.sh [hunk 2]
```diff
@@ -1,6 +1,14 @@
 #!/bin/bash
 set -v
 
+if [ gsed -v ]; then
+    gsed=gsed
+else
+    # Hopefully installed sed is the gnu one.
+    # note: aliases work poorly with xargs.
+    gsed=sed
+fi
```
Comment (Vrixyz) [inline generate_user_guides.sh:10]: using `gsed` is not trivial on linux actually, so this "hack" allows for better cross platform tooling.

**Review by sebcrozet — APPROVED** [review]
Looking fantastic, thank you!

**Pull request merged**